### ISSUE TITLE
COMPILE'ing and already compiled function gives the wrong error

### ISSUE THREAD

**Issue by drmeister** [drmeister/clasp#145]
It should just return the function

```
$ clasp_boehm_o 
Starting Boehm cclasp 0.3-test-10-255-gc4c9e3c ... loading image... it takes a few seconds
Top level.
> (defun rename-file-a-la-unix (x y)

  (let ((path    ;; (Note that the TRUENAME expression here is lifted from an
                 ;; example in the ANSI spec for TRUENAME.)
         (with-open-file (stream y :direction :output)
           (close stream)
           ;; From the ANSI spec: "In this case, the file is closed
           ;; when the truename is tried, so the truename
           ;; information is reliable."
           (truename stream))))
    (delete-file path)
    (rename-file x path)))

RENAME-FILE-A-LA-UNIX
> (compile 'rename-file-a-la-unix)

Condition of type: SIMPLE-ERROR
Could not compile func

Available restarts:
(use :r1 to invoke restart 1)

1. (RESTART-TOPLEVEL) Go back to Top-Level REPL.

Broken at frame[115] TOP-LEVEL.
 File: "/home/nyef/src/lisp/clasp/clasp-git/build/clasp/Contents/Resources/lisp/kernel/lsp/top.lsp" (Position #21032)
>> 
```

**Comment by Shinmera** [comment]
Can confirm on 
```
> (defun foo () 0)

FOO
> (compile 'foo)

Condition of type: SIMPLE-ERROR
Could not compile func
```

**Comment by Shinmera** [comment]
Confirm fixed on 0ed6a8f092150fd46552c3f1519e1ce319a085cd:
```
> (defun foo () 0)

FOO
> (compile 'foo)
;;; Warning: We have lost the original function definition for #<CLOSURE-WITH-SLOTS FOO :ftype :FUNCTION lambda-list: COMMON-LISP:NIL :fptr 0x7fa9749c42f0>. Compilation failed.

#<CLOSURE-WITH-SLOTS FOO :ftype :FUNCTION lambda-list: COMMON-LISP:NIL :fptr 0x7fa9749c42f0>
T
NIL
```
